src/controllers/queryRunner.ts

```diff
@@ -380,38 +381,81 @@ export default class QueryRunner {
             }
         }
 
+        // add the column headers
+        if (this.shouldIncludeHeaders(includeHeaders)) {
+            let firstCol: number;
+            let lastCol: number;
+            for (let range of selection) {
+                if (firstCol === undefined || (range.fromCell < firstCol)) {
+                    firstCol = range.fromCell;
+                }
+                if (lastCol === undefined || (range.toCell > lastCol)) {
+                    lastCol = range.toCell;
+                }
+            }
+            let columnRange: ISlickRange = {
+                fromCell: firstCol,
+                toCell: lastCol,
+                fromRow: undefined,
+                toRow: undefined
+            };
+            let columnHeaders = this.getColumnHeaders(batchId, resultId, columnRange);
+            copyString += columnHeaders.join('\t');
```

Comment:
Make sure you aren't having the same issue as ADS where you're appending an extra tab onto the end of the copied row

Reply:
`.join('\t')` doesn't append the joining string at the end